File: LeetCode-2023/2180.统计各位数字之和为偶数的整数个数.java
/*
 * @lc app=leetcode.cn id=2180 lang=java
 *
 * [2180] 统计各位数字之和为偶数的整数个数
 */

// @lc code=start
class Solution {
    public int countEven(int num) {
        int cnt = 0;
        for(int i = 1; i <= num; i++) {
            int sum = 0;
            int x = i;
            while(x != 0) {
                sum += x % 10;
                x /= 10;
            }
            if(sum % 2 == 0) {
                cnt++;
            }
        }
        return cnt;
    }
}
// @lc code=end

File: LeetCode-2023/2180.统计各位数字之和为偶数的整数个数.py
#
# @lc app=leetcode.cn id=2180 lang=python3
#
# [2180] 统计各位数字之和为偶数的整数个数
#

# @lc code=start
class Solution:
    def countEven(self, num: int) -> int:
        cnt = 0
        for i in range(1, num + 1):
            sum = 0
            while i != 0:
                sum = sum + (i % 10)
                i = i // 10
            if sum % 2 == 0:
                cnt = cnt + 1
        return cnt
# @lc code=end

